vnext/Microsoft.ReactNative/ReactNativeHost.idl

```diff
@@ -14,9 +14,11 @@ namespace Microsoft.ReactNative {
   runtimeclass ReactNativeHost {
     ReactNativeHost();
 
-    IVector<IReactPackageProvider> PackageProviders { get; set; };
```

Comment:
What was previous behavior of someone set this?

Reply:
I think it was buggy. It was like giving external code ability to access private variables. As a rule of thumb we must not allow the collections to be set, but rather just modify them. Why? For example, if we decide that the internal storage must be thread-safe (which is not today), then there is no way for us to enforce it if we allow the collection to be set. It means that the API has an unnecessary 'freedom' that puts us in the corner when we want to improve the implementation.

---
In reply to: [432989273](https://github.com/microsoft/react-native-windows/pull/5071#discussion_r432989273) [](ancestors = 432989273)